### PR TITLE
feat: add paper server packages

### PR DESCRIPTION
**Pull request by Misterio77**
Package all https://papermc.io servers.

### CODE REVIEW

File: pkgs/paper-servers/update.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        super().__init__(*args, **kwargs)
+
+    def send(self, request, **kwargs):
+        timeout = kwargs.get("timeout")
+        if timeout is None:
+            kwargs["timeout"] = self.timeout
+        return super().send(request, **kwargs)
+
+
+def make_client():
```
Comment (Misterio77) [inline pkgs/paper-servers/update.py:29]: I had to add some timeout and retries, as it seems the papermc api is a bit unstable (at least on my machine) and often hangs.

File: pkgs/paper-servers/update.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+            }
+
+    json.dump(output, lock, indent=2)
+    lock.write("\n")
+
+
+if __name__ == "__main__":
+    folder = Path(__file__).parent
+    lock_path = folder / "lock.json"
+    main(open(lock_path, "w"), make_client())
```
Comment (Misterio77) [inline pkgs/paper-servers/update.py:71]: I decided to write the lock from scratch every time, as it takes only about 12 seconds to do so (there's much fewer builds when compared to fabric).

File: pkgs/paper-servers/default.nix [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+      (buildNumber: value: callPackage ./derivation.nix {
+        inherit (value) url sha256;
+        version = "${mcVersion}-build.${buildNumber}";
+        jre_headless = getRecommendedJavaVersion mcVersion;
+      })
+      builds))
+    versions;
+
+  # Latest build for each MC version
+  latestBuilds = sortBy "version" versionOlder (map last packages);
```
Comment (Misterio77) [inline pkgs/paper-servers/default.nix:34]: My approach for getting the latest build for each MC version was making a nested list and grabbing the first of each. `packages` looks something like this (each element being a derivation): 
```
[ [ paper-1.17.1-build.1 paper-1.17.1-build.2 ... ] [ paper-1.18-build.1 paper-1.18-build.2 ... ] ... ]
```

File: pkgs/paper-servers/update.py (hunk 1 above)
Comment (Infinidoge) [reply pkgs/paper-servers/update.py:29]: Darn you unstable APIs!
That being said, definitely a good idea to have.

File: pkgs/paper-servers/update.py (hunk 2 above)
Comment (Infinidoge) [reply pkgs/paper-servers/update.py:71]: Makes sense, Fabric/Quilt definitely have a lot more builds with the sheer number of loader version/minecraft version combinations.

**Review by Infinidoge — CHANGES_REQUESTED** [review]
Looks good overall!

That being said, the README needs to be updated to reflect the changes before this can be merged.

Once that's done, I'll smoke test before merging.

**Pull request merged**